src/iterators/quantile.rs

```diff
@@ -61,29 +66,27 @@ impl<'a, T: 'a + Counter> PickyIterator<T> for Iter<'a, T> {
         // slice, traverse half to get to 50%. Then traverse half of the last (second) slice to get
         // to 75%, etc.
         // Minimum of 0 (1.0/1.0 = 1, log 2 of which is 0) so unsigned cast is safe.
+        // Won't hit the `inf` case because quantile < 1.0, so this should yield an actual number.
         let num_halvings = (1.0 / (1.0 - self.quantile_to_iterate_to)).log2() as u32;
         // Calculate the total number of ticks in 0-1 given that half of each slice is tick'd.
         // The number of slices is 2 ^ num_halvings, and each slice has two "half distances" to
         // tick, so we add an extra power of two to get ticks per whole distance.
         // Use u64 math so that there's less risk of overflow with large numbers of ticks and data
         // that ends up needing large numbers of halvings.
-        // TODO calculate the worst case total_ticks and make sure we can't ever overflow here
         let total_ticks = (self.ticks_per_half_distance as u64)
             .checked_mul(1_u64.checked_shl(num_halvings + 1).expect("too many halvings"))
             .expect("too many total ticks");
         let increment_size = 1.0 / total_ticks as f64;
-        self.quantile_to_iterate_to += increment_size;
+        // Unclear if it's possible for adding a very small increment to 0.999999... to yield > 1.0
+        // but let's just be safe since FP is weird and this code is not likely to be very hot.
+        self.quantile_to_iterate_to = 1.0_f64.min(self.quantile_to_iterate_to + increment_size);
         true
     }
 
     fn more(&mut self, _: usize) -> bool {
-        // We want one additional last step to 100%
-        if !self.reached_last_recorded_value && self.hist.count() != 0 {
-            self.quantile_to_iterate_to = 1.0;
-            self.reached_last_recorded_value = true;
-            true
-        } else {
-            false
-        }
```

Comment:
In the Java impl, `self.quantile_to_iterate_to` ends up being what is exposed as the quantile iterated to, rather than calculating `(accumulated count) / (total count)` at each iteration point as the Rust impl does (and also the Java impl for all iterators other than percentile). Thus, in the Java  impl, you would end up at quantile `0.998...` or similar when you ended up at the last nonzero bucket, and it was aesthetically pleasing to nudge the iterator one more slot forward to get to `1.0`.

Reply:
Hmm... maybe the Java way is better and we should expose that as well? Java uses `getPercentile()` vs `getPercentileIteratedTo()` to allow consumers to differentiate. We could expose an extra field as well in IterationValue, or apply type system shenanigans to have per-iterator value types (`<V extends IterationValue>?`) or bolted-on extra data (`IterationValue<T, V>` for some per-iterator associated type V). The upside is that it lets quantile iteration like in the dump-to-stdout example show the internal quantile value's small changes to make it clear that forward progress is happening, even when we stay at the same value for a while.

Reply:
I think having another value in `IterationValue` sounds perfectly sensible.
Not sure it'll be accessed much, but doesn't hurt to have it there.

Reply:
OK, I'll go ahead and add it. It will make the quantile iteration output a little easier to visually comprehend.